app/code/core/Mage/Sales/Model/Order.php

```diff
@@ -1248,14 +1248,15 @@ public function unhold()
 
     /**
      * Cancel order
-     *
+     * @param string $comment
+     * @param bool $graceful
      * @return $this
      */
-    public function cancel()
+    public function cancel($comment = '', $graceful = true)
```

Comment:
I think the second param `$graceful` should be removed as it has no meaning in this context, because `$this->canCancel()` can only be `true` in

https://github.com/OpenMage/magento-lts/blob/4885b20d86bba7aa200007a76620575488e25846/app/code/core/Mage/Sales/Model/Order.php#L1273-L1276

and this line 
https://github.com/OpenMage/magento-lts/blob/4885b20d86bba7aa200007a76620575488e25846/app/code/core/Mage/Sales/Model/Order.php#L1304
will never execute.

Reply:
You are so right!